runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2202,17 +2252,49 @@ bool TR_EscapeAnalysis::checkOtherDefsOfLoopAllocation(TR::Node *useNode, Candid
                TR::Node *defNode2 = _useDefInfo->getNode(defIndex2);
                TR::Node *firstChild = defNode2->getFirstChild();
                bool rhsIsHarmless = false;
-               for (Candidate *candidate = _candidates.getFirst(); candidate; candidate = candidate->getNext())
+               for (Candidate *otherAllocNode = _candidates.getFirst(); otherAllocNode; otherAllocNode = otherAllocNode->getNext())
                   {
-                  if (candidate->_node == firstChild)
+                  if (otherAllocNode->_node == firstChild)
                      {
                      rhsIsHarmless = true;
                      break;
                      }
+
+                 if (!rhsIsHarmless &&
+                    (_valueNumberInfo->getValueNumber(defNode2) == _valueNumberInfo->getValueNumber(otherAllocNode->_node)))
+                    {
+                    TR::TreeTop *treeTop;
+                    bool trackAliases = false;
+                    _aliasesOfOtherAllocNode->empty();
+                    _visitedNodes->empty();
+                    for (treeTop = otherAllocNode->_treeTop->getEnclosingBlock()->getEntry(); treeTop; treeTop = treeTop->getNextTreeTop())
+                       {
+                       TR::Node *node = treeTop->getNode();
+                       if (node->getOpCodeValue() == TR::BBEnd)
+                          break;
+
+                       collectAliasesOfAllocations(treeTop->getNode(), otherAllocNode->_node, trackAliases);
+
+                       if (treeTop == otherAllocNode->_treeTop)
+                           trackAliases = true;
+                       }
+
+                    if (/* (_valueNumberInfo->getValueNumber(defNode2) == _valueNumberInfo->getValueNumber(otherAllocNode->_node)) && */
```

Comment:
please remove dead code rather than commenting it out

Reply:
Removed dead code in 414b74b10957a8ae1b8a4426f0c29dec2a905275